.pre-commit-config.yaml

```diff
@@ -0,0 +1,98 @@
+# SPDX-FileCopyrightText:  Open Energy Transition Open Energy Transition gGmbH
+
+#
+# SPDX-License-Identifier: CC0-1.0
+
+# Copied from https://github.com/PyPSA/pypsa-eur/pull/302/files
+
+exclude: ^(LICENSES)
+
+
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v5.0.0
+  hooks:
+  - id: check-merge-conflict
+  - id: end-of-file-fixer
+  - id: fix-encoding-pragma
+  - id: mixed-line-ending
+  - id: trailing-whitespace
+#   - id: check-added-large-files
+#     args: ['--maxkb=2000']
+
+# Sort package imports alphabetically
+- repo: https://github.com/PyCQA/isort
+  rev: 6.0.1
+  hooks:
+  - id: isort
+    args: ["--profile", "black", "--filter-files"]
+
+# Find common spelling mistakes in comments and docstrings
+- repo: https://github.com/codespell-project/codespell
+  rev: v2.4.1
+  hooks:
+  - id: codespell
+    args: ['--ignore-regex="(\b[A-Z]+\b)"', '--ignore-words-list=fom,appartment,bage,ore,setis,tabacco,berfore,fo,FO']
+    types_or: [python, rst, markdown]
+    files: ^(scripts|doc)/
+
+# # Make docstrings PEP 257 compliant
+# - repo: https://github.com/PyCQA/docformatter
+#   rev: v1.7.5
+#   hooks:
+#   - id: docformatter
+#     args: ["--in-place", "--make-summary-multi-line", "--pre-summary-newline"]
+#     entry: docformatter
+#     language: python
+#     types: [python]
+
+- repo: https://github.com/keewis/blackdoc
+  rev: v0.3.9
+  hooks:
+  - id: blackdoc
+
+# Formatting with "black" coding style
```

Comment:
I recommend the `ruff` formatter and linter. It's **much** faster than black etc, and also replaces `isort`, and results in formatting that's compatible with black.